### PR TITLE
Move UTxO to Core

### PR DESCRIPTION
**Pull request by TimSheard**
Split the UTxO file in Shelley into two parts.
One that deals just with UTxO  Cardano.Ledger.UTxO and the othet which deals with what happens in the 
Shelley era, Cardano.Ledger.Shelley.UTxO

### CODE REVIEW

File: eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/Core.hs [hunk 1]
```diff
@@ -109,17 +109,17 @@ import qualified Cardano.Ledger.Shelley.Tx as Ledger
...
   ( BlockNo (..),
     Duration (..),
     SlotNo (..),
     epochInfoFirst,
     (*-),
   )
+import Cardano.Ledger.UTxO
+  ( UTxO,
+    pattern UTxO,
+  )
```
Comment (lehins) [inline eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/Core.hs]: Let's fix this while we at it: 
```suggestion
import Cardano.Ledger.UTxO (UTxO(UTxO))
```

**Review by lehins — APPROVED** [review]
This is great. Just one minor recommendation

File: eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/Core.hs (hunk 1 above)
Comment (TimSheard) [reply eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/Core.hs]: done

**Comment by TimSheard** [comment]
After some magic, managed to restore the git history to both UTxO filles.

**Review by lehins — APPROVED** [review]
Very nice!!!

File: libs/cardano-ledger-core/src/Cardano/Ledger/UTxO.hs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    DSignable c (CH.Hash (HASH c) EraIndependentTxBody)
+  ) =>
+  SafeHash c EraIndependentTxBody ->
+  KeyPair kr c ->
+  WitVKey 'Witness c
+makeWitnessVKey safe keys =
+  WitVKey (asWitness $ vKey keys) (coerce $ signedDSIGN @c (sKey keys) (extractHash safe))
+
+-- | Create witnesses for transaction
+makeWitnessesVKey ::
```
Comment (JaredCorduan) [inline libs/cardano-ledger-core/src/Cardano/Ledger/UTxO.hs:178]: this is just a side comment while I'm thinking about it, definitely ignore for now in this PR. but I think lots of the functions in this module are not really that related to the `UTxO` and should be in their own module.

**Review by JaredCorduan — APPROVED** [review]
fantastic, thank you @TimSheard !

File: libs/cardano-ledger-core/src/Cardano/Ledger/UTxO.hs (hunk 2 above)
Comment (TimSheard) [reply libs/cardano-ledger-core/src/Cardano/Ledger/UTxO.hs:178]: For the next cleanup PR

**Review by lehins — CHANGES_REQUESTED** [review]
@TimSheard Squashing all of the commits together erased all of the work you did in the attempt to preserve git history. Are you sure that is what you wanted?

**Review by lehins — APPROVED** [review]
:+1:

**Pull request merged**